### PR TITLE
Rabbitmq - Fix event sending process

### PR DESCRIPTION
**Pull request by marwanehcine**
1- fix rabbitmq event sending process after code refactor
2- adding IT test for communication between gateway and console using rabbitmq

### CODE REVIEW

File: gateway/src/test/java/org/georchestra/gateway/rabbitmq/SendMessageRabbitmqIT.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.testcontainers.shaded.org.awaitility.Awaitility.await;
+
+/**
+ * Integration tests for {@link RabbitmqAccountCreatedEventSender}.
+ */
+@SpringBootTest(classes = GeorchestraGatewayApplication.class)
+@TestPropertySource(properties = { //
+        "georchestra.datadir=${user.dir}/src/test/resources/data_directory" })
```
Comment (pmauduit) [inline gateway/src/test/java/org/georchestra/gateway/rabbitmq/SendMessageRabbitmqIT.java:55]: It does not sound to me as very portable, it doestn't resolve on my dev environment.

File: gateway/src/test/java/org/georchestra/gateway/rabbitmq/SendMessageRabbitmqIT.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        try {
+            Enumeration<NetworkInterface> networkInterfaceEnumeration = NetworkInterface.getNetworkInterfaces();
+            while (networkInterfaceEnumeration.hasMoreElements()) {
+                for (InterfaceAddress interfaceAddress : networkInterfaceEnumeration.nextElement()
+                        .getInterfaceAddresses())
+                    if (interfaceAddress.getAddress().isSiteLocalAddress())
+                        return interfaceAddress.getAddress().getHostAddress();
+            }
+        } catch (SocketException e) {
+            e.printStackTrace();
```
Comment (pmauduit) [inline gateway/src/test/java/org/georchestra/gateway/rabbitmq/SendMessageRabbitmqIT.java:143]: I'd make use of a logger, even in a test context.

**Review by pmauduit — CHANGES_REQUESTED** [review]
The GHA is failing:
https://github.com/georchestra/georchestra-gateway/pull/81/checks

so I wouldd expect the PR is not ready for review yet

**Comment by marwanehcine** [comment]
> The GHA is failing: https://github.com/georchestra/georchestra-gateway/pull/81/checks
> 
> so I wouldd expect the PR is not ready for review yet

This needs this PR to be merged before to make compilation works :
https://github.com/georchestra/georchestra/pull/4098/files

File: gateway/src/test/java/org/georchestra/gateway/rabbitmq/SendMessageRabbitmqIT.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        user.setOrganization("PSC");
+        user.setOAuth2ProviderId("testIDP");
+        eventPublisher.publishEvent(new AccountCreated(user));
+        await().atMost(30, TimeUnit.SECONDS).until(() -> {
+            return output.getOut().contains(
+                    "new OAuth2 account creation notification for testadmin@georchestra.org has been received by console");
+        });
+    }
+
+    private static String getLocalIpAddress() {
```
[the remark this answers is not in the document]
Comment (pmauduit) [reply gateway/src/test/java/org/georchestra/gateway/rabbitmq/SendMessageRabbitmqIT.java:124]: the method is unused anymore

File: gateway/pom.xml [hunk 4]
```diff
@@ -125,6 +125,14 @@
       <artifactId>json</artifactId>
       <version>20230618</version>
     </dependency>
+    <!-- testcontainers dependency added to run SendMessageRabbitmqIT
+    with rabbitmq and smtp containers.-->
```
Comment (pmauduit) [inline gateway/pom.xml:129]: maybe the comment is not necessary, I think we can assume that if the dependency is added (especially with a test scope), it's for testing purposes ;-) and the reference to smtp is a bit confusing for this dependency in particular

**Review by pmauduit — APPROVED** [review]

**Comment by pmauduit** [comment]
> This needs this PR to be merged before to make compilation works :
> https://github.com/georchestra/georchestra/pull/4098/files

I am still concerned by the failing testsuite though, the above mentioned PR has been closed without being merged, and no traces of a SmtpContainer in the georchestra/georchestra repository yet.

**Pull request merged**